### PR TITLE
Integrate database with activity views

### PR DESCRIPTION
**Pull request by jacksos101**
- Map animates to an activity that has been scanned before launching activity
- Activities save to database.

-------
Note: We still need to:
- Sort out the photograph activities save function 
- Improve the views
- Implemented the informational activity

### CODE REVIEW

**Comment by jacksos101** [comment]
Codes for testing:
[codes.zip](https://github.com/ssm-deepcove/discover_deep_cove/files/3464547/codes.zip)

File: lib/main.dart [hunk 1]
```diff
@@ -45,6 +46,7 @@ ThemeData appTheme() {
 
     textTheme: TextTheme(
       headline: TextStyle(fontSize: 30, color: Colors.white),
+      body1: TextStyle(fontSize: 16, color: Colors.white),
```
Comment (MitchellQ) [inline lib/main.dart]: Since we are setting the font size in the body widget we don't need to do so here

File: lib/data/models/activity/activity.dart [hunk 2]
```diff
@@ -94,34 +93,38 @@ class Activity {
...
-      case ActivityType.photographActivity: {
-        return userPhotoId != null;
-      }
-      break;
-      default: {
-        return false;
-      }
+  bool isCompleted() {
+    switch (activityType) {
+      case ActivityType.pictureSelectActivity:
```
Comment (MitchellQ) [inline lib/data/models/activity/activity.dart]: The opening brace should be on the same line as the case

File: lib/views/activites/picture_select_activity_view.dart [hunk 3]
```diff
@@ -62,7 +65,7 @@ class _PictureSelectActivityViewState extends State<PictureSelectActivityView> {
                     ? BodyText(
                         "You Answered:",
                       )
-                    : null,
+                    : Container(),
```
Comment (MitchellQ) [inline lib/views/activites/picture_select_activity_view.dart]: why did you change these back to empty containers

File: lib/main.dart (hunk 1 above)
Comment (MitchellQ) [reply lib/main.dart]: I will change this for you, did you find out what caused the underlines when I removed the body1 theme if we can fix that I don't see why we can't remove the theme again

Comment (MitchellQ) [reply lib/main.dart]: I have pushed changes for this

Comment (jacksos101) [reply lib/main.dart]: > I will change this for you, did you find out what caused the underlines when I removed the body1 theme if we can fix that I don't see why we can't remove the theme again

It seems to be related to not using a theme

File: lib/views/activites/picture_select_activity_view.dart (hunk 3 above)
Comment (jacksos101) [reply lib/views/activites/picture_select_activity_view.dart]: Children of columns/rows cannot be null

File: lib/main.dart (hunk 1 above)
Comment (MitchellQ) [reply lib/main.dart]: ok well we will just have to leave the color there then maybe raise an issue on the flutter repo

File: lib/data/models/activity/activity.dart (hunk 2 above)
Comment (jacksos101) [reply lib/data/models/activity/activity.dart]: Dartfmt says otherwise

Comment (MitchellQ) [reply lib/data/models/activity/activity.dart]: that's weird
